File: pursuit/accounts.api.pursuit/Account.cs
using System;

namespace accounts.api.pursuit
{
    public enum AccountType { Savings, Checking, Mortgage }

    public class Account
    {
        public Guid AccountId { get; }
        public string AccountName => $"{Users.GetUserName(UserId)}'s {AccountType}";
        public AccountType AccountType { get; }
        public decimal Balance { get; set; }
        public Guid UserId { get; }

        public Account(Guid userId, AccountType accountType, decimal balance)
        {
            UserId = userId;
            AccountType = accountType;
            Balance = balance;
            AccountId = Guid.NewGuid();
        }

    }
}

File: pursuit/mortgage.api.pursuit/src/main/kotlin/local/pursuit/api/mortgage/models/Account.kt
package local.pursuit.api.mortgage.models

import com.fasterxml.jackson.annotation.JsonValue
import java.util.*

enum class AccountType { Savings, Checking, Mortgage;
    @JsonValue
    open fun toValue(): Int {
        return ordinal
    }
}

data class Account(val accountType: AccountType, val balance: Double, val userId: UUID);
